docs/proposals/ose-rebase.md

```diff
@@ -0,0 +1,39 @@
+## Rebase OSE on Origin
+
+Today, code in OSE lands by fetching and merging Origin manually. OSE-specific changes
+are scattered in the repo and it's hard to reason about any conflicts. This doc proposes
+moving all the changes in OSE that diverge from Origin into a set of commits that
+will sit on top of the Origin code[0] and a Jenkins job that will be rebasing master and
+all active release branches on a daily basis.
+
+There are two types of commits that need to be carried on top of Origin: 
+1. long-term carries related to branding and release tooling (master and release branches)
+2. CVEs (release branches only)
+
+Rebases are going to be simpler for the master branch. The OSE master will be hard-reset to
+the Origin master and any carry commits will be applied on top. If any conflicts are hit, manual
+resolution will be required. In the future, we can extend the conflict resolution mechanism in
+Jenkins to send an e-mail either to the author and committer of the carry commit or we can store
+e-mails in the commit description and use those instead.
+
+Rebases for release branches are less trivial because they may contain backported fixes for CVEs
+that do not exist in the respective Origin branches. In such a case, backports should continue
+to be carried until we identify that the backport exists in the Origin branch and drop the OSE commit.
+How? Assuming the backport ends up w/o any additional change in Origin then cherry-picking
```

Comment:
> In the case where a bug-fix touches code that has branding, for instance, or any other part of the codebase where we carry a patch, the bug-fix as it lands in Origin will not be identical to that which hits OSE. By using a naming convention, we would not have to rely on the diffs being identical for the cherry-pick to fall out. Do we want that? I'm not sure.

When you talk about naming convention here does it apply only for OSE commits or for commits in both repos?

> I was also thinking we could just set up github/openshift/origin.git as a remote and literally git rebase onto that master branch. No need to manually cherry-pick anything. The rebase should fail in the same manner. We can hash out impls later but I think we can get porcelain output from git and not need to do any output munging.

This also sounds good.